Library/ConnectionStringParser.ts

```diff
@@ -0,0 +1,45 @@
+import { ConnectionStringFields } from "../Declarations/Contracts";
+import Constants = require("../Declarations/Constants");
+
+class ConnectionStringParser {
+    private static _FIELDS_SEPARATOR = ";";
+    private static _FIELD_KEY_VALUE_SEPARATOR = "=";
+
+    public static parse(connectionString?: string): ConnectionStringFields {
+        if (!connectionString) {
+            return {};
+        }
+
+        const kvPairs = connectionString.split(ConnectionStringParser._FIELDS_SEPARATOR);
+
+        const result: ConnectionStringFields = kvPairs.reduce((fields: ConnectionStringFields, kv: string) => {
+            const kvParts = kv.split(ConnectionStringParser._FIELD_KEY_VALUE_SEPARATOR);
+
+            if (kvParts.length === 2) { // only save fields with valid formats
+                const key = kvParts[0].toLowerCase();
+                const value = kvParts[1];
+                fields[key] = value;
+            }
+            return fields;
+        }, {});
+
+        if (Object.keys(result).length > 0) {
+            // this is a valid connection string, so parse the results
+
+            if (result.endpointsuffix) {
+                // use endpoint suffix where overrides are not provided
+                const locationPrefix = result.location ? result.location + "." : "";
```

Comment:
Can't the location prefix be used even if an endpoints suffix isn't provided?

Reply:
From what I could tell from the spec and published unit tests, it only works with endpoint suffix. @TimothyMothra to confirm

Reply:
Location will only apply to the EndpointSuffix. 

1. check for explicit endpoint (location is ignored)
2. check for endpoint suffix (location is optional)
3. use classic endpoint (location is ignored)